cli/templatevariables.go

```diff
@@ -21,7 +70,131 @@ func ParseUserVariableValues(variablesFile string, commandLineVariables []string
 		return nil, err
 	}
 
-	return combineVariableValues(fromFile, fromCommandLine), nil
+	return combineVariableValues(fromVars, fromFile, fromCommandLine), nil
+}
+
+func parseVariableValuesFromVarsFiles(varsFiles []string) ([]codersdk.VariableValue, error) {
+	var parsed []codersdk.VariableValue
+	for _, varsFile := range varsFiles {
+		content, err := os.ReadFile(varsFile)
+		if err != nil {
+			return nil, err
+		}
+
+		var t []codersdk.VariableValue
+		ext := filepath.Ext(varsFile)
+		switch ext {
+		case ".tfvars":
+			t, err = parseVariableValuesFromHCL(content)
+			if err != nil {
+				return nil, xerrors.Errorf("unable to parse HCL content: %w", err)
+			}
+		case ".json":
+			t, err = parseVariableValuesFromJSON(content)
+			if err != nil {
+				return nil, xerrors.Errorf("unable to parse JSON content: %w", err)
+			}
+		default:
+			return nil, xerrors.Errorf("unexpected tfvars format: %s", ext)
+		}
+
+		parsed = append(parsed, t...)
+	}
+	return parsed, nil
+}
+
+func parseVariableValuesFromHCL(content []byte) ([]codersdk.VariableValue, error) {
+	parser := hclparse.NewParser()
+	hclFile, diags := parser.ParseHCL(content, "file.hcl")
+	if diags.HasErrors() {
+		return nil, diags
+	}
+
+	attrs, diags := hclFile.Body.JustAttributes()
+	if diags.HasErrors() {
+		return nil, diags
+	}
+
+	stringData := map[string]string{}
+	for _, attribute := range attrs {
+		ctyValue, diags := attribute.Expr.Value(nil)
+		if diags.HasErrors() {
+			return nil, diags
+		}
+
+		ctyType := ctyValue.Type()
+		if ctyType.Equals(cty.String) {
+			stringData[attribute.Name] = ctyValue.AsString()
+		} else if ctyType.Equals(cty.Number) {
+			stringData[attribute.Name] = ctyValue.AsBigFloat().String()
+		} else if ctyType.IsTupleType() {
```

Comment:
I immediately saw this, thought 🤔 "y no switch", and then saw how annoying this is.

Reply:
Yes, unfortunately the API isn't too friendly :)